tests/retry_tests.rs

```diff
@@ -0,0 +1,56 @@
+#[cfg(test)]
+mod retry_tests {
+	use std::sync::atomic::{AtomicU32, Ordering};
+	use std::sync::Arc;
+	use std::time::Duration;
+
+	use vss_client::error::VssError;
+	use vss_client::util::retry::{retry, ExponentialBackoffRetryPolicy, RetryPolicy};
+
+	#[tokio::test]
+	async fn test_async_retry() {
+		let base_delay = Duration::from_millis(10);
+		let max_attempts = 3;
+		let max_total_delay = Duration::from_secs(60);
+		let max_jitter = Duration::from_millis(5);
+
+		let exponential_backoff_jitter_policy = ExponentialBackoffRetryPolicy::new(base_delay)
+			.skip_retry_on_error(|e| matches!(e, VssError::InvalidRequestError(..)))
+			.with_max_attempts(max_attempts)
+			.with_max_total_delay(max_total_delay)
+			.with_max_jitter(max_jitter);
```

Comment:
In practice, this test is exercising:
- `Some` case of `skip_retry_on_error`
- `Some` and `None` cases for `with_max_attempts`
- `Some` case of `with_max_total_delay`

We should be able to test:
- `None` case for `skip_retry_on_error`
- `None` case for `with_max_total_delay`

You may want to make separate tests with their own retry policy construction so that it is clear what is being tested.

Anything else with time would probably to be difficult or too flaky to worry about.

Reply:
Not the absence of max_total_delay, but rather when the accumulated delay is sufficient enough for `MaxTotalDelayRetryPolicy` to return `None`.